core/src/main/java/org/elasticsearch/common/breaker/CircuitBreakingException.java

```diff
@@ -67,6 +68,11 @@ public long getByteLimit() {
     }
 
     @Override
+    public RestStatus status() {
+        return RestStatus.INTERNAL_SERVER_ERROR;
```

Comment:
To me I think the rule should be: we override this method if you have a specific rest http status code that's associated with the exception. I this case we just use a generic code that's not really indicative of the error. However, looking at alternative, I suggest we use 503 Service Unavailable